### PR TITLE
make submission id show as default over core job id. Fix download logs button not downloading full logs

### LINKED ISSUE #40467: Downloading Dashboard Logs only have 50k lines
### What happened + What you expected to happen

https://anyscaleteam.slack.com/archives/CKD613X8S/p1697663877306529

the dashboard tells me "[Truncation warning] This log has been truncated and only the latest 50000 lines are displayed. Click "Download" button above to see the full log" but when I download it's still truncated

### Versions / Dependencies

Ray: 2.7.1

### Reproduction script

Repro: Under aviary org, https://session-7gr8b45tsbxyrvv8e5lstvm7fu.i.anyscaleuserdata.com/#/actors/a139fede134c2fd2b9c45e9223000000

### Issue Severity

None

### PR DESCRIPTION
**Pull request by alanwguo**
## Why are these changes needed?
For people who use job submission, they are more likely to know about job submission ids than core job ids. So if a job submission id exists, we should show those first.

When downloading logs, we should enable downloading the full logs and not a truncated amount of logs.

Screenshot 2023-10-23 at 3 04 15 PM

## Related issue number
fixes #40467 

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by rkooo567 — COMMENTED** [review]
Can't we just have both?

**Comment by rkooo567** [comment]
Looks like I need frontend owner to approve this?

**Comment by alanwguo** [comment]
> Can't we just have both?

hard to fit in the UI. Do you think both is better? I guess I can do raysubmit_12345 (01000000)

File: dashboard/client/src/service/log.ts [hunk 1]
```diff
@@ -66,6 +66,10 @@ export type StateApiLogInput = {
    * If filename is provided, suffix is not necessary
    */
   filename?: string | null;
+  /**
+   * -1 for all lines.
```
Comment (shawnpanda) [inline dashboard/client/src/service/log.ts:70]: nit: my recollection of typescript typing is that the comment `-1 for all lines.` is an implementation detail and should be separate from the type declaration.

**Review by shawnpanda — APPROVED** [review]

File: dashboard/client/src/service/log.ts (hunk 1 above)
Comment (alanwguo) [reply dashboard/client/src/service/log.ts:70]: I want this to be be part of the docs for this parameter so people who use this component know they can pass in -1

**Comment by rkooo567** [comment]
I cannot merge this now. I assume we need an approval from frontend owner?

**Comment by alanwguo** [comment]
@rkooo567 , perhaps it just needs someone from ray team to approve? Shawn has approved but maybe he does not have the right permissions? Can you approve given Shawn has reviewed the code?

**Review by rkooo567 — APPROVED** [review]

**Pull request merged**